pkg/controller/core/workload_controller.go

```diff
@@ -210,8 +214,11 @@ func (r *WorkloadReconciler) reconcileSyncAdmissionChecks(ctx context.Context, w
 		log := ctrl.LoggerFrom(ctx)
 		log.V(3).Info("The workload needs admission checks updates", "clusterQueue", klog.KRef("", cqName), "admissionChecks", queueAdmissionChecks)
 		wl.Status.AdmissionChecks = newChecks
-		err := r.client.Status().Update(ctx, wl)
-		return true, client.IgnoreNotFound(err)
+		if err := r.client.Status().Update(ctx, wl); !apierrors.IsNotFound(err) {
+			return true, err
+		}
+		r.record.Eventf(wl, corev1.EventTypeNormal, "UpdatedWorkload", "Updated admission checks of the workload %v for clusterQueue %s", workload.Key(wl), cqName)
```

Comment:
Also the event name is a bit generic

Reply:
+1 `WorkloadAdmissionChecksUpdated` ?

Reply:
changed https://github.com/kubernetes-sigs/kueue/pull/1271/commits/99243245c6f0e4573eefe1e6b60840aec6dae92e